rusoto/core/src/xmlerror.rs

```diff
@@ -14,37 +14,38 @@ impl XmlErrorDeserializer {
     pub fn deserialize<T: Peek + Next>(tag_name: &str,
                                        stack: &mut T)
                                        -> Result<XmlError, XmlParseError> {
-        try!(start_element(tag_name, stack));
+        start_element(tag_name, stack)?;
```

Comment:
Any reason we changed from: `try!()`, to: `?`. I'm generally more of a fan of try unless it's nested very deep because it's explicit to someone who hasn't perhaps worked with rust as much.

I don't think it really needs to be changed back since it's not that big of a deal, I'm just curious why it was chosen.

Reply:
My understanding is that `try!` is deprecated. From the [Rust API Documentation](https://doc.rust-lang.org/nightly/std/macro.try.html), "The `?` operator was added to replace `try!` and should be used instead." Also, as far as I can see the [second edition of The Rust Programming Language](https://doc.rust-lang.org/book/second-edition/ch09-02-recoverable-errors-with-result.html) no longer mentions `try!` and I expect other intros to Rust to follow.

Reply:
Ah I didn't realize they had fully deprecated it. I guess I didn't watch that PR to closely 😲

Reply:
Huh, I didn't know Rust best practices are moving to `?`.  Especially when https://github.com/rust-lang/rust/issues/37939 is still unresolved. But it's probably the least of Rusoto's performance concerns. 😉